router.go

```diff
@@ -0,0 +1,133 @@
+package main
+
+import (
+	"net/http"
+)
+
+type Router struct {
+	name   string
+	routes []*Route
+}
+
+// func (r *Router) Query(opId string, fn interface{}) *Route {
+// 	// todo: should I return error?
+// 	inpType, outType, err := validateAndRetrieveHandlerParamType(fn)
+// 	if err != nil {
+// 		panic(err)
+// 	}
+
+// 	handler := &Handler{inpType, outType, fn}
+// 	route := &Route{
+// 		handler: handler,
+// 		routeConfig: &RouteConfig{
+// 			opId:   opId,
+// 			opType: "query",
+// 			method: http.MethodGet,
+// 			url:    "/v1/" + opId,
+// 			jsonSchema: &JSONSchemaForInputAndOutput{
+// 				input:  jsonschema.ReflectFromType(inpType),
+// 				output: jsonschema.ReflectFromType(outType),
+// 			},
+// 		}}
+// 	r.routes = append(r.routes, route)
+// 	return route
+// }
+
+func (r *Router) Query(items ...interface{}) *Route {
+	var operationId string
+	var function interface{}
+	var route *Route
+	// if the Query function is called with operationID
+	if len(items) >= 1 {
+		operationId = items[0].(string)
+		route = &Route{
+			handler: &Handler{},
+			routeConfig: &RouteConfig{
+				opId:       operationId,
+				opType:     "query",
+				method:     http.MethodGet,
+				url:        "/v1/" + operationId,
+				jsonSchema: &JSONSchemaForInputAndOutput{},
+			}}
+	}
+
+	// if the function is called alongwith the Handler fn
+	if len(items) == 2 {
+		function = items[1]
+		route = route.fn(function)
+
+	} else if len(items) < 1 || len(items) > 2 {
+		panic("Illegal number of arguments provided to Query")
+	}
+	r.routes = append(r.routes, route)
+	return route
+}
+
+func (r *Router) Mutation(items ...interface{}) *Route {
+	var operationId string
+	var function interface{}
+	var route *Route
+	// if the Query function is called with operationID
+	if len(items) >= 1 {
+		operationId = items[0].(string)
+		route = &Route{
+			handler: &Handler{},
+			routeConfig: &RouteConfig{
+				opId:       operationId,
+				opType:     "mutation",
+				method:     http.MethodPost,
+				url:        "/v1/" + operationId,
+				jsonSchema: &JSONSchemaForInputAndOutput{},
```

Comment:
I suggest you calculare the jsonSchema here itself

Reply:
If the Mutation() or Query() only accepts 1 parameter, i.e. 'name', then I will not get a handler function.
In such case, my routeConfig would have an empty object for jsonSchema.

Reply:
Yes. You are correct. Lets calculate the jsonschema in the Fn() method then which will accept the user's function

Reply:
Got it. Will do that.